lasagne/tests/layers/test_recurrent.py

```diff
@@ -59,6 +59,50 @@ def test_recurrent_nparams_learn_init():
     assert len(lasagne.layers.get_all_params(l_rec, regularizable=False)) == 2
 
 
+def test_recurrent_hid_init_layer():
+    # test that you can set hid_init to be a layer
+    l_inp = InputLayer((2, 2, 3))
+    l_inp_h = InputLayer((2, 5))
+    l_rec = RecurrentLayer(l_inp, 5, hid_init=l_inp_h)
+
+    x = T.tensor3()
+    h = T.matrix()
+
+    output = lasagne.layers.get_output(l_rec, {l_inp: x, l_inp_h: h})
+
+
+def test_recurrent_nparams_hid_init_layer():
+    # test that you can see layers through hid_init
+    l_inp = InputLayer((2, 2, 3))
+    l_inp_h = InputLayer((2, 5))
+    l_inp_h_de = DenseLayer(l_inp_h, 7)
+    l_rec = RecurrentLayer(l_inp, 7, hid_init=l_inp_h_de)
+
+    x = T.tensor3()
+    h = T.matrix()
+
+    # b, W_hid_to_hid and W_in_to_hid + W + b
+    assert len(lasagne.layers.get_all_params(l_rec, trainable=True)) == 5
```

Comment:
I would first add a test directly checking whether you can see the layers through hid_init:
```
assert lasagne.layers.get_all_layers(l_rec) == [l_inp, l_inp_h, l_inp_h_de, l_rec]
```
Your test only checks whether the parameters of `hid_init` itself are included, not whether you can fully "see through" it to the incoming layers of `hid_init`.